test/audiovideocontroller/DefaultAudioVideoController.test.ts

```diff
@@ -1746,4 +1767,203 @@ describe('DefaultAudioVideoController', () => {
+      );
+      const attendees = audioVideoController.getRemoteVideosAvailable();
+      expect(attendees).to.have.lengthOf(0);
+    });
+  });
+
+  describe('remoteVideosAvailableDidChange observer', () => {
+    it('observer should get called only when added', async () => {
+      audioVideoController = new DefaultAudioVideoController(
+        configuration,
+        new NoOpDebugLogger(),
+        webSocketAdapter,
+        new NoOpMediaStreamBroker(),
+        reconnectController
+      );
+      let event = 0;
+      let observed = 0;
+      class TestObserver implements AudioVideoObserver {
+        remoteVideosAvailableDidChange(
+          _attendees: { attendeeId: string; externalUserId: string }[]
+        ): void {
+          if (event !== 1 && event !== 3 && event !== 5) {
+            assert.fail();
+          }
+          observed += 1;
+        }
+      }
+      const observer = new TestObserver();
+      await delay();
+      audioVideoController.addObserver(observer);
+      event += 1;
+      audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
+        observer.remoteVideosAvailableDidChange([]);
+      });
+      await delay();
+      audioVideoController.removeObserver(observer);
+      event += 1;
+      audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
+        observer.remoteVideosAvailableDidChange([]);
+      });
+      await delay();
+      audioVideoController.addObserver(observer);
+      event += 1;
+      audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
+        observer.remoteVideosAvailableDidChange([]);
+      });
+      await delay();
+      event += 1;
+      audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
+        observer.remoteVideosAvailableDidChange([]);
+      });
+      audioVideoController.removeObserver(observer);
+      await delay();
+      audioVideoController.addObserver(observer);
+      event += 1;
+      audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
+        observer.remoteVideosAvailableDidChange([]);
+      });
+      await delay();
+      expect(event).to.equal(5);
+      expect(observed).to.equal(3);
+    });
+  });
+
+  describe('addSimulcastUplinkPolicyObserver', () => {
+    it('adds SimulcastObserver in videoUplinkBandwidthPolicy', async () => {
+      configuration.enableUnifiedPlanForChromiumBasedBrowsers = true;
+      configuration.enableSimulcastForUnifiedPlanChromiumBasedBrowsers = true;
+      domMockBehavior.browserName = 'chrome';
+      domMockBuilder = new DOMMockBuilder(domMockBehavior);
+      audioVideoController = new DefaultAudioVideoController(
+        configuration,
+        new NoOpDebugLogger(),
+        webSocketAdapter,
+        new NoOpDeviceController(),
+        reconnectController
+      );
+      await start();
+      // @ts-ignore
+      const policy: SimulcastUplinkPolicy = audioVideoController.meetingSessionContext
```

Comment:
Don't need to make changes. Just some tips for future PRs:

- Try avoiding to access a private property in test files (I understand we've done in several places...)
- Also, I'm not sure `adds SimulcastObserver in videoUplinkBandwidthPolicy` is needed because you are already testing if a simculast observer method is called.

   Similarly for the remove test, you can just test if the obsever is not called.

Reply:
This is required since I am no where testing for the actual `addSimulcastUplinkPolicyObserver` function in `DefaultAudioVideoController`. Similarly for remove one. Please correct me if I am miss understanding this.

Reply:
Have removed `addSimulcastUplinkPolicyObserver` and `removeSimulcastUplinkPolicyObserver` tests and methods.